app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+
+    # check if user is admin and give admin full privellage
+    if user.role == 'admin'
+      can :manage, :all
```

Comment:
- [ ] [optional] Instead of checking the role directly here, I would suggest creating the below method inside the user model and calling it here `user.admin?` to check the admin privilege. It will be a cleaner way of checking the admin privilege.

```ruby
 def admin?
     role == 'admin' ? true : false
   end
```